sycl/unittests/program_manager/passing_link_and_compile_options.cpp

```diff
@@ -0,0 +1,529 @@
+  PiArray<PiProperty> ImgKPOI{std::move(EAMKernelPOI)};
+
+  PiPropertySet PropSet;
+  PropSet.insert(__SYCL_PI_PROPERTY_SET_KERNEL_PARAM_OPT_INFO,
+                 std::move(ImgKPOI));
+
+  std::vector<unsigned char> Bin{0, 1, 2, 3, 4, 5}; // Random data
+
+  PiArray<PiOffloadEntry> Entries = makeEmptyKernels({EAMTestKernelName2});
+
+  PiImage Img{PI_DEVICE_BINARY_TYPE_SPIRV,            // Format
+              __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64, // DeviceTargetSpec
+              _cmplOptions,                           // Compile options
+              _lnkOptions,                            // Link options
+              std::move(Bin),
+              std::move(Entries),
+              std::move(PropSet)};
+  return Img;
+}
+
+inline pi_result redefinedProgramLink(pi_context, pi_uint32, const pi_device *,
+                                      const char *_linkOpts, pi_uint32,
+                                      const pi_program *,
+                                      void (*)(pi_program, void *), void *,
+                                      pi_program *) {
+  assert(_linkOpts != nullptr);
+  current_link_options = std::string(_linkOpts);
+  return PI_SUCCESS;
+}
+
+inline pi_result redefinedProgramCompile(pi_program, pi_uint32,
+                                         const pi_device *,
+                                         const char *_compileOpts, pi_uint32,
+                                         const pi_program *, const char **,
+                                         void (*)(pi_program, void *), void *) {
+  assert(_compileOpts != nullptr);
+  current_compile_options = std::string(_compileOpts);
+  return PI_SUCCESS;
+}
+
+inline pi_result redefinedProgramBuild(
+    pi_program prog, pi_uint32, const pi_device *, const char *options,
+    void (*pfn_notify)(pi_program program, void *user_data), void *user_data) {
+  return PI_SUCCESS;
+}
+
+TEST(Link_Compile_Options, compile_link_Options_Test_empty_options) {
+  sycl::platform Plt{sycl::default_selector()};
+  if (Plt.is_host()) {
+    std::cerr << "Test is not supported on host, skipping\n";
+    return; // test is not supported on host.
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_cuda) {
+    std::cerr << "Test is not supported on CUDA platform, skipping\n";
+    return;
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_hip) {
+    std::cerr << "Test is not supported on HIP platform, skipping\n";
+    return;
+  }
+  sycl::unittest::PiMock Mock{Plt};
+  setupDefaultMockAPIs(Mock);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramCompile>(
+      redefinedProgramCompile);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramLink>(redefinedProgramLink);
+
+  const sycl::device Dev = Plt.get_devices()[0];
+  // Check devices' compile & link options separately
+  current_link_options.clear();
+  current_compile_options.clear();
+  std::string expected_compile_options1 = "", expected_compile_options2 = "";
```

Comment:
Do we really need several strings here? It seems the test is about empty options, so one "expected" string would be enough?

Reply:
You are right. Using two empty string is unnecessary here.